util/src/main/java/io/kubernetes/client/Configmap.java

```diff
@@ -0,0 +1,203 @@
+import java.nio.file.Path;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+import okhttp3.Call;
+import okhttp3.Response;
+
+/** Utility class to create Configmaps */
+public class Configmap {
+
+  private ApiClient apiClient;
+  private CoreV1Api coreClient;
+
+  public Configmap() {
+    this(Configuration.getDefaultApiClient());
+  }
+
+  public Configmap(ApiClient apiClient) {
+    this.apiClient = apiClient;
+    this.coreClient = new CoreV1Api(apiClient);
+  }
+
+  public ApiClient getApiClient() {
+    return apiClient;
+  }
+
+  public void setApiClient(ApiClient apiClient) {
+    this.apiClient = apiClient;
+  }
+
+  /**
+   * Method to create a Configmap from a file. The file should contain key-value pairs
+   *
+   * @param namespace
+   * @param name
+   * @param file
+   * @throws IOException
+   * @throws CreateConfigmapException
+   * @throws ApiException
+   */
+  public void createFromFile(String namespace, String name, Path file)
+      throws IOException, CreateConfigmapException, ApiException {
+    if (Files.isRegularFile(file)) {
+      try (BufferedReader bufferedReader =
+          new BufferedReader(new FileReader(String.valueOf(file)))) {
+        createFromReader(namespace, name, bufferedReader);
+      }
+    } else {
+      throw new CreateConfigmapException("Target path is a directory");
+    }
+  }
+
+  /**
+   * To create Configmap by passing InputStream of a file which has key value pairs
+   *
+   * @param namespace
+   * @param name
+   * @param stream
+   * @throws IOException
+   * @throws CreateConfigmapException
+   * @throws ApiException
+   */
+  public void createFromStream(String namespace, String name, InputStream stream)
+      throws IOException, CreateConfigmapException, ApiException {
+    createFromReader(namespace, name, new InputStreamReader(stream));
+  }
+
+  /**
+   * To creat Configmap by passing string literal. Value should be a string in the form "key=value"
+   *
+   * @param namespace
+   * @param name
+   * @param value
+   * @throws IOException
+   * @throws CreateConfigmapException
+   * @throws ApiException
+   */
+  public void createFromString(String namespace, String name, String value)
+      throws IOException, CreateConfigmapException, ApiException {
+    createFromReader(namespace, name, new StringReader(value));
+  }
+
+  /**
+   * To create Configmap by passing Reader of a file which has key value pairs
+   *
+   * @param namespace
+   * @param name
+   * @param reader
+   * @throws IOException
+   * @throws CreateConfigmapException
+   * @throws ApiException
+   */
+  public void createFromReader(String namespace, String name, Reader reader)
+      throws IOException, CreateConfigmapException, ApiException {
+    if (reader != null) {
+      if (name != null && !name.isEmpty()) {
+        V1ConfigMap v1ConfigMap = createConfigmapObject(name);
+        byte[] data = CharStreams.toString(reader).getBytes();
+        if (isValidUtf8(data)) {
```

Comment:
Does `kubectl` do this? I thought everything was base64 encoded, even if it was already valid utf-8.

Reply:
This was as i understood from the Kubectl code.  BTW i am a noob in Golang.

Reply:
Line number 262 at https://github.com/kubernetes/kubectl/blob/master/pkg/generate/versioned/configmap.go#L262